<a href="https://colab.research.google.com/github/kiq-caique/inteligencia_artificial_UFV/blob/main/Machine_Learn_Atividade_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1 - Crie uma variável relacionando duas outras variáveis do dataset, como mostrado em exemplo na aula síncrona (sugestões de variáveis: número de famílias, total de quartos e total de cômodos) e calcule o coeficiente de correlação linear entre ela e a variável alvo.

In [ ]:
import os

## NumPy para manipular matrizes e vetores
import numpy as np

## Pandas para manipular os dataframes
import pandas as pd


In [ ]:
# baixar o arquivo do GitHub
# o repositorio alvo eh o repositorio da disciplina
# !wget https://github.com/rodolpho-neves/elt574-aprendizado-de-maquina/raw/main/semana-1/housing.csv

--2024-07-02 19:37:27--  https://github.com/rodolpho-neves/elt574-aprendizado-de-maquina/raw/main/semana-1/housing.csv
Resolving github.com (github.com)... 140.82.112.4
Connecting to github.com (github.com)|140.82.112.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/rodolpho-neves/elt574-aprendizado-de-maquina/main/semana-1/housing.csv [following]
--2024-07-02 19:37:27--  https://raw.githubusercontent.com/rodolpho-neves/elt574-aprendizado-de-maquina/main/semana-1/housing.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.109.133, 185.199.111.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1423529 (1.4M) [text/plain]
Saving to: ‘housing.csv’

housing.csv         100%[===================>]   1.36M  --.-KB/s    in 0.09s   

2024-07-02 19:37:28 (14.7 MB/s) - ‘hous

In [ ]:
path = '/content/melb_data.csv'

df_housing = pd.read_csv(path)
print('Dataset carregado!')

# Removendo colunas qualitativas que não serão utilizadas no exercício
df_housing = df_housing.drop([ 'BuildingArea', 'YearBuilt','Suburb', 'Address', 'Type', 'Method', 'SellerG', 'Date', 'CouncilArea', 'Regionname'], axis=1)
df_housing.head()

Dataset carregado!


Rooms      Price  Distance  Postcode  Bedroom2  Bathroom  Car  Landsize  \
0      2  1480000.0       2.5    3067.0       2.0       1.0  1.0     202.0   
1      2  1035000.0       2.5    3067.0       2.0       1.0  0.0     156.0   
2      3  1465000.0       2.5    3067.0       3.0       2.0  0.0     134.0   
3      3   850000.0       2.5    3067.0       3.0       2.0  1.0      94.0   
4      4  1600000.0       2.5    3067.0       3.0       1.0  2.0     120.0   

   Lattitude  Longtitude  Propertycount  
0   -37.7996    144.9984         4019.0  
1   -37.8079    144.9934         4019.0  
2   -37.8093    144.9944         4019.0  
3   -37.7969    144.9969         4019.0  
4   -37.8072    144.9941         4019.0

In [ ]:
#removendo NAs relativos às colunas restantes
df_housing = df_housing.dropna()

# Conferindo a matrix de correlação atual
corr_matrix = df_housing.corr(numeric_only=True)
corr_matrix

Rooms     Price  Distance  Postcode  Bedroom2  Bathroom  \
Rooms          1.000000  0.497656  0.293479  0.055551  0.944021  0.592917   
Price          0.497656  1.000000 -0.161081  0.107371  0.476846  0.467720   
Distance       0.293479 -0.161081  1.000000  0.433515  0.295296  0.126343   
Postcode       0.055551  0.107371  0.433515  1.000000  0.060825  0.113885   
Bedroom2       0.944021  0.476846  0.295296  0.060825  1.000000  0.584657   
Bathroom       0.592917  0.467720  0.126343  0.113885  0.584657  1.000000   
Car            0.408483  0.238979  0.262994  0.050289  0.405325  0.322246   
Landsize       0.025609  0.037482  0.025209  0.024899  0.025574  0.037160   
Lattitude      0.016123 -0.212444 -0.132830 -0.406699  0.016101 -0.070666   
Longtitude     0.100427  0.204093  0.239363  0.446873  0.101918  0.118594   
Propertycount -0.081394 -0.041673 -0.055744  0.061987 -0.081206 -0.051733   

                    Car  Landsize  Lattitude  Longtitude  Propertycount  
Rooms          0.408483  0.025609   0.016123    0.100427      -0.081394  
Price          0.238979  0.037482  -0.212444    0.204093      -0.041673  
Distance       0.262994  0.025209  -0.132830    0.239363      -0.055744  
Postcode       0.050289  0.024899  -0.406699    0.446873       0.061987  
Bedroom2       0.405325  0.025574   0.016101    0.101918      -0.081206  
Bathroom       0.322246  0.037160  -0.070666    0.118594      -0.051733  
Car            1.000000  0.026770  -0.001963    0.063395      -0.024295  
Landsize       0.026770  1.000000   0.009550    0.010837      -0.006746  
Lattitude     -0.001963  0.009550   1.000000   -0.357743       0.047141  
Longtitude     0.063395  0.010837  -0.357743    1.000000       0.066484  
Propertycount -0.024295 -0.006746   0.047141    0.066484       1.000000

In [ ]:
# Correlação entre Rooms e Distance
df_housing["rooms_distance"] = df_housing["Rooms"]/df_housing["Distance"]

#correlações
corr_matrix=df_housing.corr()
corr_matrix['Price'].sort_values(ascending=False)

Price             1.000000
Rooms             0.497656
Bedroom2          0.476846
Bathroom          0.467720
rooms_distance    0.347204
Car               0.238979
Longtitude        0.204093
Postcode          0.107371
Landsize          0.037482
Propertycount    -0.041673
Distance         -0.161081
Lattitude        -0.212444
Name: Price, dtype: float64

#2 - A variável criada possui algum grau de correlação linear relevante com a variável alvo? O que pode ser explicado por esta correlação?

Sim, possui impacto moderado.
A correlação das variáveis rooms_distance e Price é de 0.3472 e indica que a relação não é extremamente forte, mas ainda assim é significativa. Portanto, o aumento na variável criada não garante um aumento proporcionalmente igual no Preço do imóvel, mas há uma influência positiva observável.